TEMPLATES.md

```diff
@@ -0,0 +1,103 @@
+# Using `Template`s as Execution Targets With CI Operator
+
+The CI Operator allows for users to configure tests running in a single container
+with the `tests` array. In many cases, however, running a command in a single
+container will not suffice. For more complicated tests, the CI Operator supports
+a `--template` option with which a test author may write black-boxed execution
+targets using OpenShift `Template`s. This document explains the interfaces expected
+for test `Template`s and best practices for how an author should configure their
+`Template`.
+
+The high-level contract for a `Template` execution target is that the `Template`
+will be processed into the test's `Namespace` once prerequisite `Build`s have
+occurred. Artifacts will be gathered from `artifacts` volumes on `Pod`s in
+the `Template`, if any are present. What work occurs in a `Template`'s `Pod`s is
+a black box as far as the CI Operator is concerned.
+
+## Parameters Available to `Template`s
+
+A number of parameters are available to the `Template` and will be provided if
```

Comment:
This sentence is confusing to me. Unless I'm wrong, isn't it just simpler to say "The following template variables are available:"? As it stands, this sentence seems to suggest that they won't be available unless the Template explicitly asks for them, which doesn't sound like any templating system I've ever heard of.